Pool_Properties_Design.md

```diff
@@ -0,0 +1,192 @@
+
+## Motivation
+
+As we all know, we did a lot of work to design OpenSDS Profile (see [here](https://docs.google.com/document/d/1irNnz019j0XuW6SZNigs6QuYFOCC3uL44EkRYJiHyq8/edit?usp=sharing)). In addition, the extensible [capabilties filter](https://github.com/opensds/opensds/pull/297) mechanism was also finished several days ago. This design spec proposes how to implement the StoragePool resources based on the Swordfish spec.
+
+Now that we have come to a conclusion about `Profile` design, it would be much easier to design `StoragePool` because both of them are composed of LineOfService defined in Swordfish.
+
+## Proposal
+
+### Goals
+
+* Redesign the definition of `ExtraSpec` contained in `StoragePoolSpec`.
+
+* Update all storage drivers to stay up with the changes.
+
+* Update the fake pool data declared in test package.
+
+### Data model impact
+
+This proposal will update the definition of `StoragePoolSpec` with these changes below:
+
+* Implement `DataStorageLoS`, `IOConnectivityLoS` and `DataProtectionLos` struct which defined in Swordfish:
+```go
+// DataStorageLoS can be used to describe a service option covering storage
+// provisioning and availability.
+type DataStorageLoS struct {
+	// The enumeration literal specifies the time after a disaster that the
+	// client shall regain conformant service level access to the primary
+	// store, currently it only supports "OnlineActive","OnlinePassive",
+	// "Nearline" and "Offline".
+	// The expectation is that the services required to implement this
+	// capability are part of the advertising system.
+	RecoveryTimeObjective int64 `json:"recoveryTimeObjective,omitempty" yaml:"recoveryTimeObjective,omitempty"`
+
+	// ProvisioningPolicy only supports "Fixed" and "Thin".
+	ProvisioningPolicy string `json:"provisioningPolicy,omitempty" yaml:"provisioningPolicy,omitempty"`
+
+	// IsSpaceEfficient indicates that the storage is compressed or deduplicated.
+	// The default value for this prperty is false.
+	IsSpaceEfficient bool `json:"isSpaceEfficient,omitempty" yaml:"isSpaceEfficient,omitempty"`
+}
+
+// IOConnectivityLoS can be used to specify the characteristics of storage
+// connectivity.
+type IOConnectivityLoS struct {
+	// The Enumeration Literal shall specify the Access protocol for this
+	// service option.
+	AccessProtocol string `json:"accessProtocol,omitempty" yaml:"accessProtocol,omitempty"`
+
+	// MaxIOPS shall be the maximum IOs per second that the connection shall
+	// allow for the selected access protocol.
+	// +units:[IO]/s
+	MaxIOPS int64 `json:"maxIOPS,omitempty" yaml:"maxIOPS,omitempty"`
```

Comment:
On a second thought, I think we should add MaxBWS here as that is the feedback from Kei.

Reply:
Sure